### ISSUE TITLE
`str(sdreport(obj))` throws an error due to `as.list.sdreport` default `what = ""`

### ISSUE THREAD

**Issue by jaganmn** [kaskr/adcomp#352]
```r
library("TMB")
packageVersion("TMB") # 1.7.20
R.version.string # 4.1.0
runExample("simple")
sdr <- sdreport(obj)
```

`str(sdr)` complains:

```
Error in match.arg(what, opts) : 
  'arg' should be one of “Estimate”, “Std. Error”
```

`traceback()` gives:

```
8: stop(gettextf("'arg' should be one of %s", paste(dQuote(choices), 
       collapse = ", ")), domain = NA)
7: match.arg(what, opts)
6: as.list.sdreport(X)
5: as.list(X)
4: vapply(object, typeof, "")
3: "promise" %in% (oTypes <- vapply(object, typeof, ""))
2: str.default(sdr)
1: str(sdr)
```

Presumably, the error would be avoided if the default value of `as.list.sdreport` argument `what` were changed from the empty string `""` to one of the two valid options `"Estimate"` and `"Std. Error"`.  An alternative would be to export a simple method for `str`:

```r
str.sdreport <- function(object, ...) {
  object <- unclass(object)
  NextMethod("str")
}
```

Now `str(sdr)` runs as expected:

```
> str(sdr)
List of 10
 $ value          : Named num [1:2] 1.03 141.71
  ..- attr(*, "names")= chr [1:2] "sd0" "sum(exp(u))"
 $ sd             : num [1:2] 0.069 15.598
 $ cov            : num [1:2, 1:2] 0.00476 -0.42417 -0.42417 243.30429
 $ par.fixed      : Named num [1:4] 52.0137 30.2406 -0.1578 0.0333
  ..- attr(*, "names")= chr [1:4] "beta" "beta" "logsdu" "logsd0"
 $ cov.fixed      : num [1:4, 1:4] 0.040677 0.017933 0.000938 -0.000399 0.017933 ...
  ..- attr(*, "dimnames")=List of 2
  .. ..$ : chr [1:4] "beta" "beta" "logsdu" "logsd0"
  .. ..$ : chr [1:4] "beta" "beta" "logsdu" "logsd0"
 $ pdHess         : logi TRUE
 $ gradient.fixed : num [1:4] 2.73e-06 4.16e-06 -6.06e-06 -7.41e-06
 $ par.random     : Named num [1:114] -0.481 -0.717 0.657 -0.218 0.416 ...
  ..- attr(*, "names")= chr [1:114] "u" "u" "u" "u" ...
 $ diag.cov.random: num [1:114] 0.324 0.328 0.326 0.32 0.322 ...
 $ env            :<environment: 0x121adcd30> 
```

**Comment by bbolker** [comment]
duplicate of https://github.com/glmmTMB/glmmTMB/issues/692, resolved in  fd49c10  (not that there's any way you could have known)

**Comment by jaganmn** [comment]
Oops - thanks.
